src/main/scala/org/constellation/util/HealthChecker.scala

```diff
@@ -70,106 +55,63 @@ case class RecentSync(hash: String, height: Long)
-      peersSnapshots <- collectSnapshot(peers)
-      _ <- clearStaleTips(peersSnapshots)
-
-      major <- majorityStateChooser
-        .chooseMajorityState(peersSnapshots :+ (dao.id, ownSnapshots), maxOrZero(ownSnapshots))
-        .getOrElse((Seq[RecentSnapshot](), Set[Id]()))
-
-      diff <- compareSnapshotState(major, ownSnapshots)
-
-//      result <- Sync[F].pure[Option[List[RecentSnapshot]]](None)
-      result <- if (shouldReDownload(ownSnapshots, diff)) {
-        logger.info(
-          s"[${dao.id.short}] Re-download process with : " +
-            s"Snapshot to download : ${diff.snapshotsToDownload} " +
-            s"Snapshot to delete : ${diff.snapshotsToDelete} " +
-            s"From peers : ${diff.peers} " +
-            s"Own snapshots : $ownSnapshots " +
-            s"Major state : $major"
-        ) >>
-          startReDownload(diff, peers.filterKeys(diff.peers.contains))
-            .flatMap(_ => Sync[F].delay[Option[List[RecentSnapshot]]](Some(major._1.toList)))
-      } else {
-        Sync[F].pure[Option[List[RecentSnapshot]]](None)
-      }
-    } yield result
+      _ <- clearStaleTips()
+    } yield ()
 
     check.recoverWith {
       case err =>
         logger
-          .error(err)(s"[${dao.id.short}] Unexpected error during re-download process: ${err.getMessage}")
-          .flatMap(_ => Sync[F].pure[Option[List[RecentSnapshot]]](None))
+          .error(err)(s"[${dao.id.short}] Unexpected error during check for stale tips process: ${err.getMessage}")
+          .flatMap(_ => Sync[F].pure[Unit](None))
     }
   }
 
-  private[util] def compareSnapshotState(major: (Seq[RecentSnapshot], Set[Id]), ownSnapshots: List[RecentSnapshot]) =
-    SnapshotDiff(
-      ownSnapshots.diff(major._1).sortBy(_.height).reverse,
-      major._1.diff(ownSnapshots).toList.sortBy(_.height).reverse,
-      major._2.toList
-    ).pure[F]
-
-  def clearStaleTips(clusterSnapshots: List[(Id, List[RecentSnapshot])]): F[Unit] = {
-    val nodesWithHeights = clusterSnapshots.filter(_._2.nonEmpty)
-    if (clusterSnapshots.size - nodesWithHeights.size < dao.processingConfig.numFacilitatorPeers && nodesWithHeights.size >= dao.processingConfig.numFacilitatorPeers) {
-      val maxHeightsOfMinimumFacilitators = nodesWithHeights
-        .map(x => x._2.map(_.height).max)
-        .groupBy(x => x)
-        .filter(t => t._2.size >= dao.processingConfig.numFacilitatorPeers)
-
-      if (maxHeightsOfMinimumFacilitators.nonEmpty)
-        concurrentTipService.clearStaleTips(
-          maxHeightsOfMinimumFacilitators.keySet.min + snapshotHeightInterval
+  private def collectMinTipHeights(): F[Map[Id, Long]] =
+    for {
+      peers <- LiftIO[F].liftIO(dao.readyPeers(NodeType.Full))
+      minTipHeights <- peers.values.toList
+        .map(_.client)
+        .traverse(_.getNonBlockingF[F, (Id, Long)]("heights/min")(calculationContext))
+    } yield minTipHeights.toMap
+
+  def clearStaleTips(): F[Unit] =
```

Comment:
Lets change this name since we already have a clearStaleTips method on concurrentTipService which actually does the removal